Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/DualShockTests.cs

```diff
@@ -175,9 +175,16 @@ public void Devices_SupportsDualShockOnPS4()
         Assert.That(gamepad.acceleration.y.ReadValue(), Is.EqualTo(0.654).Within(0.00001));
         Assert.That(gamepad.acceleration.z.ReadValue(), Is.EqualTo(0.321).Within(0.00001));
 
-        Assert.That(gamepad.orientation.x.ReadValue(), Is.EqualTo(0.111).Within(0.00001));
-        Assert.That(gamepad.orientation.y.ReadValue(), Is.EqualTo(0.222).Within(0.00001));
-        Assert.That(gamepad.orientation.z.ReadValue(), Is.EqualTo(0.333).Within(0.00001));
+        Quaternion orientation = gamepad.orientation.ReadValue();
```

Comment:
Sidenote: reading a quaternion as a whole is not necessarily equivalent to reading its individual x, y, z, and w controls (which is one reason I prefer reading individual controls in tests; you don't want to test the combination into a quaternion here but rather just ensure the controls have the expected value). Combined controls often end up with processors on them which then alter the values. E.g. the stick controls have deadzone processor on them so reading `leftStick` directly yields a different value than doing `new Vector2(leftStick.x.ReadValue(), leftStick.y.ReadValue())`. Anyway, don't think it's overly relevant here.

Reply:
Not sure what you are getting at here. I know I could't find any individual components on the QuaternionControl. The comment in that class confirmed that adding in individual quaternion fields wasn't correct for a Quaternion, which I totally understand. 

However I'm not sure the best way to test this. Should the test create a quaternion from Euler angles and then use ReadValue() to get the quaternion and convert that back into  Euler angles and then test those?

Or should it just be removed if its not a relevant test anyway.

Reply:
Ah doh, you're right, that didn't connect properly in my brain. I didn't realize that QuaternionControl indeed works different that way from Vector3Control. So, all good.